### PR TITLE
fix tainted-sql-string FP from sentiment

### PR DESCRIPTION
**Pull request by kurt-r2c**
- add FP test case
- remove overly permissive regex
- add find `conditions => ...` matcher

### CODE REVIEW

File: ruby/rails/security/injection/tainted-sql-string.yaml [hunk 1]
```diff
@@ -1,54 +1,61 @@
...
-        - focus-metavariable: $EXPR
-        - metavariable-regex:
-            metavariable: $SQLSTR
-            regex: (?i)(select|delete|insert|create|update|alter|drop)\b
+            - patterns:
+                - pattern-either:
+                    - pattern: |
+                        $RECORD.where($X,...)
+                    - pattern: |
+                        $RECORD.find(..., :conditions => $X,...)
```
Comment (LewisArdern) [inline ruby/rails/security/injection/tainted-sql-string.yaml:40]: should this be => ...,$X,...

File: ruby/rails/security/injection/tainted-sql-string.yaml [hunk 2]
```diff
@@ -1,54 +1,61 @@
...
+                - focus-metavariable: $X
+            - patterns:
+                - pattern: |
+                    "...#{$EXPR}..."
+                - pattern-not-inside: |
+                    $FUNC("...",  "...#{$EXPR}...",...)
```
Comment (LewisArdern) [inline ruby/rails/security/injection/tainted-sql-string.yaml:46]: Should this be metavariable-regex with:
```"$SELECT#{$EXPR}...",...)```

File: ruby/rails/security/injection/tainted-sql-string.yaml [hunk 3]
```diff
@@ -1,54 +1,61 @@
...
+            - patterns:
+                - pattern-either:
+                    - pattern: |
+                        $RECORD.where($X,...)
+                    - pattern: |
+                        $RECORD.find(..., :conditions => $X,...)
+                - focus-metavariable: $X
+            - patterns:
+                - pattern: |
+                    "...#{$EXPR}..."
```
Comment (LewisArdern) [inline ruby/rails/security/injection/tainted-sql-string.yaml]: Should this be metavariable-regex with:
```"$SELECT...#{$EXPR}...```

**Review by LewisArdern — APPROVED** [review]

File: ruby/rails/security/injection/tainted-sql-string.yaml (hunk 1 above)
Comment (kurt-r2c) [reply ruby/rails/security/injection/tainted-sql-string.yaml:40]: the sink is the `conditions` string - it lets you specify a partial raw SQL query. `$Y.find(...)` is SELECT FROM WHERE <conditions> under the hood: `User.find(:first, :conditions => "user_id = '#{params[:user][:user_id]}'")`

File: ruby/rails/security/injection/tainted-sql-string.yaml (hunk 3 above)
Comment (kurt-r2c) [reply ruby/rails/security/injection/tainted-sql-string.yaml]: Might not matter, but made this change and tests pass so 🤷  might as well

**Pull request merged**